src/kubernetes.cc

```diff
@@ -169,45 +265,26 @@ MetadataUpdater::ResourceMetadata KubernetesReader::GetPodMetadata(
   const std::string namespace_name = metadata->Get<json::String>("namespace");
   const std::string pod_name = metadata->Get<json::String>("name");
   const std::string pod_id = metadata->Get<json::String>("uid");
-  const std::string pod_version = pod->Get<json::String>("apiVersion");
-  const std::string pod_type = "io.k8s.Pod";
-  const std::string pod_schema =
-      format::Substitute(kKubernetesSchemaNameFormat,
-                         {{"type", pod_type}, {"version", pod_version}});
 
   const MonitoredResource k8s_pod("k8s_pod", {
     {"cluster_name", cluster_name},
     {"namespace_name", namespace_name},
     {"pod_name", pod_name},
     {"location", cluster_location},
   });
-
-  if (config_.VerboseLogging()) {
-    LOG(INFO) << "Raw pod metadata: " << *pod;
-  }
-
   const std::string k8s_pod_id = boost::algorithm::join(
       std::vector<std::string>{kK8sPodResourcePrefix, pod_id},
       config_.MetadataApiResourceTypeSeparator());
   const std::string k8s_pod_name = boost::algorithm::join(
       std::vector<std::string>{kK8sPodResourcePrefix, namespace_name, pod_name},
       config_.MetadataApiResourceTypeSeparator());
-  const std::string pod_full_name =
-      format::Substitute(kPodFullNameFormat,
-                         {{"cluster_full_name", ClusterFullName()},
-                          {"namespace_name", namespace_name},
-                          {"pod_name", pod_name}});
+
+  MetadataStore::Metadata pod_metadata =
+      GetMetadata(pod, collected_at, is_deleted);
   return MetadataUpdater::ResourceMetadata(
       std::vector<std::string>{k8s_pod_id, k8s_pod_name},
       k8s_pod,
-#ifdef ENABLE_KUBERNETES_METADATA
-      MetadataStore::Metadata(pod_full_name, pod_type, cluster_location,
-                              pod_version, pod_schema,
-                              is_deleted, collected_at, pod->Clone())
-#else
-      MetadataStore::Metadata::IGNORED()
-#endif
-  );
+      pod_metadata.Clone());
 }
 
 MetadataUpdater::ResourceMetadata KubernetesReader::GetContainerMetadata(
```

Comment:
We forgot to delete this, didn't we? Probably makes sense to send a separate cleanup PR that blows away container metadata and makes `MetadataQuery` behave the same as the watcher threads...